spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -171,20 +164,27 @@ def compute_snrs(waveform_extractor, peak_sign: str = 'neg', peak_mode: str = "e
     snrs : dict
         Computed signal to noise ratio for each unit.
     """
+    if waveform_extractor.is_extension("noise_levels"):
+        noise_levels = waveform_extractor.load_extension("noise_levels").get_data()
+    else:
+        assert waveform_extractor.has_recording(), \
+            ("SNR cannot be computed because waveform_extractor is recordingless and "
+             "'noise_levels' extension is not available")
+        if random_chunk_kwargs_dict is None:
+            random_chunk_kwargs_dict = {}
+        noise_levels = get_noise_levels(waveform_extractor.recording,
+                                        return_scaled=waveform_extractor.return_scaled,
+                                        **random_chunk_kwargs_dict)
```

Comment:
Cool. I wanted to make a separate PR to do this at many palces.
Lets keep this here for now.